src/porepy/geometry/constrain_geometry.py

```diff
@@ -483,13 +483,21 @@ def polygons_by_polyhedron(
             axis=0,
         )
         # Uniquify intersection coordinates, and update the segments
-        unique_coords, _, ib = pp.utils.setmembership.uniquify_point_set(
+        unique_coords, _, ib = pp.array_operations.uniquify_point_set(
             coord_extended, tol=tol
         )
         unique_segments = ib[segments]
+
         # Then uniquify the segments, in terms of the unique coordinates
-        unique_segments, *rest = pp.utils.setmembership.uniquify_point_set(
-            unique_segments
+
+        # TODO>EK: Here, it is logical to call the int counterpart of this function.
+        # But historicaly the float version was called, which returns the same result,
+        # but ordered differently. For some reason, this ordering is important, and
+        # TestMixedDimensionalUpwinding.test_3d_2d_1d_0d otherwise fails.
```

Comment:
See comment elsewhere: Please use the logical function, update reference value in test.

Reply:
This is the only place with the test to be changed

Reply:
I updated reference values, please double-check that we didn't break anything.

Reply:
I reverted this change. The reference values remain the same